pkg/transaction/metric/metricspublisher.go

```diff
@@ -67,13 +68,30 @@ func (c *metricPublisher) publishToLighthouse(event LighthouseUsageEvent) error
 		c.logger.WithError(err).Error("publishing usage")
 		return err
 	}
-	if response.Code >= 400 {
-		resBody := string(response.Body)
+
+	if response.Code < 400 {
+		return nil
+	} else if response.Code >= 500 {
+		err := fmt.Errorf("Server error")
+		c.logger.WithField("statusCode", response.Code).WithError(err).Error(string(response.Body))
+		return err
+	}
+
+	usageResp := LighthouseUsageResponse{}
+	unmarshalErr := json.Unmarshal(response.Body, &usageResp)
+	if unmarshalErr != nil {
+		c.logger.WithField("responseBody", string(response.Body)).WithField("statusCode", response.Code).Error(unmarshalErr)
```

Comment:
update this to 

c.logger.WithField...WithField...WithError(unmarshalErr).Error("helpful message")